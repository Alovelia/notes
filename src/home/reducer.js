import { constants, actions } from 'ducks-helpers';
import handleActions from 'redux-actions/lib/handleActions';
import _ from 'common/helpers';

export const TYPE = constants('home', [
  '~GET',
  '~GET_NOTES',
  //†type
]);
export const ACTION = actions(TYPE);

// The initial state
export const initialState = _.fromJS({
  folders: [],
  notes: [],
});

export default handleActions({
  [TYPE.GET_SUCCESS]: getSuccess,
  [TYPE.GET_NOTES_SUCCESS]: getNotesSuccess,
  //†handler
}, initialState);

function getSuccess(state, { payload }) {
  return state.set('folders', _.fromJS(payload));
}
function getNotesSuccess(state, { payload }) {
  return state.set('notes', _.fromJS(payload));
}
//†reducer
