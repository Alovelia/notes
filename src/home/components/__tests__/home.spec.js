import { shallow } from 'enzyme';
import React from 'react';
import noop from 'lodash/noop';
import Home from '../layout/home';
import { ACTION } from '../../reducer';

describe('<Home />', () => {
  let wrapper;
  let stub;
  beforeAll(() => {
    stub = sinon.stub(ACTION, 'get').returns({ type: 'mock' });
  });
  afterAll(() => {
    stub.restore();
  });
  it('when initializing, the expected output is rendered', () => {
    const props = {
      trigger: noop
    };
    wrapper = shallow(<Home {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
