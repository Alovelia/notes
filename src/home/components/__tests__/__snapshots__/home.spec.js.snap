// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> when initializing, the expected output is rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeComponent
    trigger={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="folders"
        >
          <Connect(FoldersContainer) />
        </div>,
        <div
          className="notes"
        />,
      ],
      "className": "clearfix",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(FoldersContainer) />,
          "className": "folders",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "className": "notes",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="folders"
          >
            <Connect(FoldersContainer) />
          </div>,
          <div
            className="notes"
          />,
        ],
        "className": "clearfix",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(FoldersContainer) />,
            "className": "folders",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": null,
            "className": "notes",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
